modules/SofaGeneralAnimationLoop/MechanicalMatrixMapper.inl

```diff
@@ -0,0 +1,518 @@
+        }
+    }
+    Keig.setFromTriplets(tripletList.begin(), tripletList.end());
+}
+template<class InputFormat>
+static void copyMappingJacobianToEigenFormat(const typename InputFormat::MatrixDeriv& J, Eigen::SparseMatrix<double>& Jeig)
+{
+    typedef typename InputFormat::MatrixDeriv::RowConstIterator RowConstIterator;
+    typedef typename InputFormat::MatrixDeriv::ColConstIterator ColConstIterator;
+    typedef typename InputFormat::Deriv Deriv;
+    int DerivSize = InputFormat::Deriv::total_size;
+    int nbRowsJ = Jeig.rows();
+    int maxRowIndex = 0, maxColIndex = 0;
+    std::vector< Eigen::Triplet<double> > tripletListJ;
+
+    for (RowConstIterator rowIt = J.begin(); rowIt !=  J.end(); ++rowIt)
+    {
+        int rowIndex = rowIt.index();
+        if (rowIndex>maxRowIndex)
+            maxRowIndex = rowIndex;
+        for (ColConstIterator colIt = rowIt.begin(); colIt !=  rowIt.end(); ++colIt)
+        {
+            int colIndex = colIt.index();
+            Deriv elemVal = colIt.val();
+            for (int i=0;i<DerivSize;i++)
+            {
+                tripletListJ.push_back(Eigen::Triplet<double>(rowIndex,DerivSize*colIndex + i,elemVal[i]));
+                if (colIndex>maxColIndex)
+                        maxColIndex = colIndex;
+            }
+        }
+    }
+    Jeig.resize(nbRowsJ,DerivSize*(maxColIndex+1));
+    Jeig.reserve(J.size());
+    Jeig.setFromTriplets(tripletListJ.begin(), tripletListJ.end());
+}
+template<class DataTypes1, class DataTypes2>
+void MechanicalMatrixMapper<DataTypes1, DataTypes2>::optimizeAndCopyMappingJacobianToEigenFormat1(const typename DataTypes1::MatrixDeriv& J, Eigen::SparseMatrix<double>& Jeig)
+{
+    copyMappingJacobianToEigenFormat<DataTypes1>(J, Jeig);
+}
+
+template<class DataTypes1, class DataTypes2>
+void MechanicalMatrixMapper<DataTypes1, DataTypes2>::optimizeAndCopyMappingJacobianToEigenFormat2(const typename DataTypes2::MatrixDeriv& J, Eigen::SparseMatrix<double>& Jeig)
+{
+    copyMappingJacobianToEigenFormat<DataTypes2>(J, Jeig);
+}
+
+template<class DataTypes1, class DataTypes2>
+void MechanicalMatrixMapper<DataTypes1, DataTypes2>::addMassToSystem(const MechanicalParams* mparams, const DefaultMultiMatrixAccessor* KAccessor)
+{
+    if (l_mappedMass != NULL)
+    {
+        l_mappedMass->addMToMatrix(mparams, KAccessor);
+    }
+    else
+    {
+        msg_info(this) << "There is no mappedMass";
+    }
+}
+
+
+template<class DataTypes1, class DataTypes2>
+void MechanicalMatrixMapper<DataTypes1, DataTypes2>::addPrecomputedMassToSystem(const MechanicalParams* mparams, const unsigned int mstateSize,const Eigen::SparseMatrix<double> &Jeig, Eigen::SparseMatrix<double> &JtKJeig)
+{
+    SOFA_UNUSED(mparams);
+    SOFA_UNUSED(mstateSize);
+    SOFA_UNUSED(Jeig);
+    SOFA_UNUSED(JtKJeig);
+}
+
+
+
+
+template<class DataTypes1, class DataTypes2>
+void MechanicalMatrixMapper<DataTypes1, DataTypes2>::addKToMatrix(const MechanicalParams* mparams,
```

Comment:
This one is quite a big process. Do you think it could be possible to add a test for this method with known results?